### PR TITLE
ci(all): Enforce using Conventional Commit PR title

### PR DESCRIPTION
**Pull request by miquelbeltran**
## Description

Enforces the use of Conventional Commits on all PRs.

This ensures that all work can be managed using `melos version`, as it requires following Conventional Commits strictly.

Also makes reading PR changes easier.

I used the example provided by https://github.com/marketplace/actions/semantic-pull-request to configure it.

## Related Issues

None

## Checklist

- [x] I read the Contributor Guide and followed the process outlined there for submitting PRs.
- [x] I titled the PR using [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0).
- [x] I did not modify the `CHANGELOG.md` nor the `pubspec.yaml` files.
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate that with a `!` in the title as explained in [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0)).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by vbuberen** [comment]
I think we need to move this check into a separate workflow, because `all_plugins` runs not for all commits (see `ignored-paths`), while we would want to be sure in all incoming commits.

Also, I see that the referenced workflow uses https://github.com/marketplace/actions/semantic-pull-request action and I would just use it directly.

File: .github/workflows/pr.yml
Comment (miquelbeltran) [inline .github/workflows/pr.yml:8]: This workflow will run on PR changes, not code changes

File: .github/workflows/pr.yml
Comment (miquelbeltran) [inline .github/workflows/pr.yml:18]: I am not sure if this is required, I copied it over from the example

**Comment by miquelbeltran** [comment]
@vbuberen I have addressed your points. I am unable to check if this really works as it doesn't seem to trigger if it is in a branch and not on main, so we would have to wait until merged to see it in action.

**Comment by mhadaily** [comment]
This is great

**Review by nohli — APPROVED** [review]
Great idea, let's see if it works 😊

**Review by vbuberen — APPROVED** [review]

**Pull request merged**
